DataRepo/models/peak_group_label.py

```diff
@@ -0,0 +1,219 @@
+                fields=["peak_group", "element"],
+                name="unique_peakgrouplabel",
+            )
+        ]
+
+    def __str__(self):
+        return str(f"{self.element}")
+
+    @property  # type: ignore
+    @cached_function
+    def measured_compound(self):
+        """
+        Returns the first measured compound of the peakgroup containing the supplied element
+        """
+        measured_compound = None
+        measured_compounds = self.peak_group.compounds.all()
+        for compound in measured_compounds:
+            if compound.atom_count(self.element) > 0:
+                measured_compound = compound
+                break
+        return measured_compound
+
+    @property  # type: ignore
+    @cached_function
+    def enrichment_fraction(self):
+        """
+        A weighted average of the fraction of labeled atoms for this PeakGroup
+        in this sample with this (labeled) element.
+        i.e. The fraction of carbons that are labeled in this.PeakGroup compound
+        Sum of all (PeakData.fraction * PeakData.labeled_count) /
+            PeakGroup.Compound.num_atoms(PeakData.labeled_element)
+        """
+        from DataRepo.models.peak_data import PeakData
+        from DataRepo.models.peak_data_label import PeakDataLabel
+
+        enrichment_fraction = None
+        compound = None
+        error = False
+        msg = ""
+
+        try:
+            # This assumes that multiple measured compounds for the same PeakGroup are composed of the same elements
+            compound = self.measured_compound
+            if compound is None:
+                raise NoCommonLabel(self)
+            # Calculate the numerator
+            element_enrichment_sum = 0.0
+            label_pd_recs = self.peak_group.peak_data.filter(
+                labels__element__exact=self.element
+            )
+            # This assumes that if there are any label_pd_recs for this measured elem, the calculation is valid
+            if label_pd_recs.count() == 0:
+                raise PeakData.DoesNotExist()
+            for label_pd_rec in label_pd_recs:
+                # This assumes the PeakDataLabel unique constraint: peak_data, element
+                label_rec = label_pd_rec.labels.get(element__exact=self.element)
+                # And this assumes that label_rec must exist because of the filter above the loop
+                element_enrichment_sum = element_enrichment_sum + (
+                    label_pd_rec.fraction * label_rec.count
+                )
+
+            # Calculate the denominator
+            atom_count = compound.atom_count(self.element)
+
+            enrichment_fraction = element_enrichment_sum / atom_count
+
+        except (AttributeError, TypeError, NoCommonLabel) as e:
+            error = True
+            # The last 2 should not happen since the fields in PeakDataLabel are null=False, but to hard against
```

Comment:
Not sure what was intended here, but perhaps:
```suggestion
            # The last 2 should not happen since the fields in PeakDataLabel are null=False, but to guard against
```

Reply:
- [ ] Second hard/guard typo